crates/re_log_types/src/component_types/tensor.rs

```diff
@@ -828,21 +835,81 @@ impl Tensor {
 
 #[cfg(feature = "image")]
 impl Tensor {
+    #[cfg(not(target_arch = "wasm32"))]
+    /// Construct a tensor from the contents of an image file on disk.
+    ///
+    /// JPEGs will be kept encoded, left to the viewer to decode on-the-fly.
+    /// Other images types will be decoded directly.
+    ///
+    /// Requires the `image` feature.
+    #[cfg(not(target_arch = "wasm32"))]
+    pub fn from_image_file(path: &std::path::Path) -> Result<Self, TensorImageLoadError> {
+        crate::profile_function!(path.to_string_lossy());
+
+        let img_bytes = {
+            crate::profile_scope!("fs::read");
+            std::fs::read(path)?
+        };
+
+        let img_format = if let Some(extension) = path.extension() {
+            image::ImageFormat::from_extension(extension).ok_or_else(|| {
+                TensorImageLoadError::UnknownExtension {
```

Comment:
find it a bit strange that we wouldn't try to guess anyways. Many applications are fine with a `.sdflkjsd` file if the FOURCC bytes are still good